src/network/network_survey.cpp

```diff
@@ -0,0 +1,409 @@
+
+		if ((c->palette & GRFP_GRF_MASK) == GRFP_GRF_UNSET) grf["palette"] = "unset";
+		if ((c->palette & GRFP_GRF_MASK) == GRFP_GRF_DOS) grf["palette"] = "dos";
+		if ((c->palette & GRFP_GRF_MASK) == GRFP_GRF_WINDOWS) grf["palette"] = "windows";
+		if ((c->palette & GRFP_GRF_MASK) == GRFP_GRF_ANY) grf["palette"] = "any";
+
+		if ((c->palette & GRFP_BLT_MASK) == GRFP_BLT_UNSET) grf["blitter"] = "unset";
+		if ((c->palette & GRFP_BLT_MASK) == GRFP_BLT_32BPP) grf["blitter"] = "32bpp";
+
+		std::vector<std::string> flags;
+		if (HasBit(c->flags, GCF_SYSTEM)) flags.push_back("system");
+		if (HasBit(c->flags, GCF_UNSAFE)) flags.push_back("unsafe");
+		if (HasBit(c->flags, GCF_STATIC)) flags.push_back("static");
+		if (HasBit(c->flags, GCF_COMPATIBLE)) flags.push_back("compatible");
+		if (HasBit(c->flags, GCF_COPY)) flags.push_back("copy");
+		if (HasBit(c->flags, GCF_INIT_ONLY)) flags.push_back("init_only");
+		if (HasBit(c->flags, GCF_RESERVED)) flags.push_back("reserved");
+		if (HasBit(c->flags, GCF_INVALID)) flags.push_back("invalid");
+		grf["flags"] = flags;
+
+		std::vector<uint32> parameters;
+		for (int i = 0; i < c->num_params; i++) {
+			parameters.push_back(c->param[i]);
+		}
+		grf["parameters"] = parameters;
+	}
+}
+
+/**
+ * Convert game-script information to JSON.
+ *
+ * @param survey The JSON object.
+ */
+static void SurveyGameScript(nlohmann::json &survey)
+{
+	if (Game::GetInfo() == nullptr) return;
+
+	survey = fmt::format("{}.{}", Game::GetInfo()->GetName(), Game::GetInfo()->GetVersion());
+}
+
+#endif /* WITH_NLOHMANN_JSON */
+
+/**
+ * Create the payload for the survey.
+ *
+ * @param reason The reason for sending the survey.
+ * @param for_preview Whether the payload is meant for preview. This indents the result, and filters out the uuid/key.
+ * @return std::string The JSON payload as string for the survey.
+ */
+std::string NetworkSurveyHandler::CreatePayload(Reason reason, bool for_preview)
+{
+#ifndef WITH_NLOHMANN_JSON
+	return "";
+#else
+	nlohmann::json survey;
+
+	survey["reason"] = reason;
+
+	/* uuid and key are used to authenticate surveys results and are not meant
+	 * to be previewed by the user. */
+	if (!for_preview) {
+		survey["uuid"] = _savegame_uuid;
+#ifdef SURVEY_KEY
+		survey["key"] = SURVEY_KEY;
+#else
+		survey["key"] = "";
+#endif
+	}
+
+	{
+		auto &info = survey["info"];
+		SurveyOS(info["os"]);
+		info["os"]["hardware_concurrency"] = std::thread::hardware_concurrency();
+
+		SurveyOpenTTD(info["openttd"]);
+		SurveyConfiguration(info["configuration"]);
+		SurveyFont(info["font"]);
+	}
+
+	{
+		auto &game = survey["game"];
+		game["ticks"] = TimerGameTick::counter;
+		game["time"] = std::chrono::duration_cast<std::chrono::seconds>(std::chrono::steady_clock::now() - _switch_mode_time).count();
+		SurveyCompanies(game["companies"]);
+		SurveySettings(game["settings"]);
+		SurveyGrfs(game["grfs"]);
+		SurveyGameScript(game["game_script"]);
+	}
+
+	return survey.dump(for_preview ? 4 : -1);
```

Comment:
It's not obvious what the 4 or -1 do from the code.
`static const int PREVIEW_INDENT_LEVEL = 4;` or something similar?